libs/sdk-common/src/input_parser.rs

```diff
@@ -175,6 +175,27 @@ pub async fn parse(input: &str) -> Result<InputType> {
         };
     }
 
+    // Covers BIP 21 URIs and simple onchain Liquid addresses (which are valid BIP 21 with the 'liquid:' or 'liquidtestnet:' prefix)
+    if let Ok(liquid_address_data) = parse_liquid_bip21(input) {
+        // Special case of LN BOLT11 with onchain fallback
+        // Search for the `lightning=bolt11` param in the BIP21 URI and, if found, extract the bolt11
+        let mut invoice_param: Option<LNInvoice> = None;
+        if let Some(query) = input.split('?').collect::<Vec<_>>().get(1) {
+            invoice_param = querystring::querify(query)
+                .iter()
+                .find(|(key, _)| key == &"lightning")
+                .map(|(_, value)| parse_invoice(value))
+                .transpose()?;
+        }
+
+        return match invoice_param {
```

Comment:
This is different than bip21 for the greenlight sdk. Here we truly want to return the parsed bip21 (both invoice and address). Perhaps also in the greenlight sdk we should have done it this way.
I am thinking of the other info that the sdk might use (label, etc...). Also the input parser is a generic parser that can be used by the user regardless of payments.
How about returning here LiquidBip21 structure?

Reply:
Will do! I left the invoice as plain string, though we do parse it in the bip21 parser. Should we return `LNInvoice` struct directly instead?

Reply:
I think plain string is just fine. User an parse it if desired.